lib/plugins/wallet-scoring/ciphertrace/ciphertrace.js

```diff
@@ -38,12 +39,54 @@ function isValidWalletScore(account, score) {
   const client = getClient(account)
   if (_.isNil(client)) return Promise.resolve(true)
 
-  const threshold = _.isNil(account) ? WALLET_SCORE_THRESHOLD : account.scoreThreshold
-  return Promise.resolve(score < threshold)
+  const threshold = _.isNil(account.scoreThreshold) ? WALLET_SCORE_THRESHOLD : account.scoreThreshold
+  return _.isNil(account) ? Promise.resolve(true) : Promise.resolve(score < threshold)
+}
+
+function getTransactionHash(account, cryptoCode, receivingAddress) {
+  const client = getClient(account)
+  if (!_.includes(_.toUpper(cryptoCode), SUPPORTED_COINS) || _.isNil(client)) return Promise.resolve(null)
+
+  const { apiVersion, authHeader } = client
+
+  return axios.get(`https://rest.ciphertrace.com/api/${apiVersion}/${_.toLower(cryptoCode) !== 'btc' ? `${_.toLower(cryptoCode)}_` : ``}address/search?features=tx&address=${receivingAddress}`, {
+    headers: authHeader
+  })
+    .then(res => {
+      const data = res.data
+      if (_.size(data.txHistory) > 1) {
+        logger.warn('An address generated by this wallet was used in more than one transaction')
+      }
+      return _.join(', ', _.map(it => it.txHash, data.txHistory))
```

Comment:
Is the return value supposed to be really the join of the different hashes, the array of them, or something else?

Because this result is eventually used by `getInputAddresses()` in the request URL.